qualtran/bloqs/qft/approximate_qft.py

```diff
@@ -0,0 +1,121 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import math
+from functools import cached_property
+
+import attrs
+import cirq
+import numpy as np
+from attr import field
+from numpy.typing import NDArray
+
+from qualtran import GateWithRegisters, QFxp, QUInt, Signature
+from qualtran.bloqs.arithmetic.multiplication import PlusEqualProduct
+
+
+@attrs.frozen
+class ApproximateQFT(GateWithRegisters):
```

Comment:
Can you use the auto doc infrastructure to generate a corresponding tutorial notebook? You can see my recent PR on how to do this when adding a new Bloq

https://github.com/quantumlib/Qualtran/commit/6b6900f71a2f226fa5b8ee6557bcb3073897edcd

Specifically, do the following
1. Create functions in `approximate_qft.py` file that instantiate an aqft Bloq and decorate it with a `@bloq_example`
2. Create a `_CC_AQFT_DPC = BloqDocSpec` object and include it in the `autogenerate-bloqs-notebooks-v2.py` file
3. Use function from step-1 in a bloq_autotester test in `approximate_qft_test.py`
4. Run `python dev_tools/autogenerate-bloqs-notebooks-v2.py` to generate a tutorial notebook, verify the latex renders correctly and commit it.